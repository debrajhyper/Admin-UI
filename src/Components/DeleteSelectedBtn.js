import React from 'react'

import { FaTrash } from 'react-icons/fa';

const DeleteSelectedBtn = ({ users, setUsers }) => {

    //  Delete the selected users from the users array and set the users to the state using the setUsers() method.
    const handleDeleteSelected = e => {
        e.preventDefault();
        let newUsers = [...users];
        newUsers = newUsers.filter(user => {
            return !user.isChecked;
        });
        setUsers(newUsers);
    };
    
    return (
        <button 
            data-testid="delete-selected-btn"
            id="first" 
            type="button"
            title="Delete selected"
            className="flex items-center justify-center p-2 md:px-4 shadow-md shadow-red-300 dark:shadow-red-900 text-white bg-red-500 transition-all duration-500 rounded focus:shadow hover:bg-red-600 hover:shadow-lg hover:shadow-red-200 outline-none focus:outline-none" 
            onClick={handleDeleteSelected} 
        >
            <FaTrash />
            <span className="ml-1 md:ml-2 text-xs md:text-base text-left">
                Delete Selected
            </span>
        </button>
    )
}

export default DeleteSelectedBtn
