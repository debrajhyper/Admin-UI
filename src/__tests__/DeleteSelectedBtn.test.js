import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import DeleteSelectedBtn from "../Components/DeleteSelectedBtn";

afterEach(cleanup);

test("render without crashing", () => {
    render(<DeleteSelectedBtn />);

    const BtnElement = screen.getByTestId("delete-selected-btn");
    expect(BtnElement).toBeInTheDocument();
    expect(BtnElement).toHaveAttribute("title", "Delete selected");
    expect(BtnElement).toHaveTextContent("Delete Selected");
})

test("snapshot", () => {
    const tree = renderer.create(<DeleteSelectedBtn />).toJSON();
    expect(tree).toMatchSnapshot();
})