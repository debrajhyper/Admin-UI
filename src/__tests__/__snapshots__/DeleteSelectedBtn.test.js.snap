// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<button
  className="flex items-center justify-center p-2 md:px-4 shadow-md shadow-red-300 dark:shadow-red-900 text-white bg-red-500 transition-all duration-500 rounded focus:shadow hover:bg-red-600 hover:shadow-lg hover:shadow-red-200 outline-none focus:outline-none"
  data-testid="delete-selected-btn"
  id="first"
  onClick={[Function]}
  title="Delete selected"
  type="button"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
    />
  </svg>
  <span
    className="ml-1 md:ml-2 text-xs md:text-base text-left"
  >
    Delete Selected
  </span>
</button>
`;
